llm/llama.go

```diff
@@ -544,11 +544,18 @@ func (llm *llama) Predict(ctx context.Context, prevContext []int, prompt string,
 		Stop:             llm.Stop,
 	}
 
-	data, err := json.Marshal(predReq)
-	if err != nil {
-		return fmt.Errorf("error marshaling data: %v", err)
+	// Handling JSON marshaling with special characters unescaped.
+	buf := &bytes.Buffer{}
+	enc := json.NewEncoder(buf)
+	enc.SetEscapeHTML(false)
+
+	if err := enc.Encode(predReq); err != nil {
+		return fmt.Errorf("failed to marshal data: %v", err)
 	}
 
+	// Remove trailing newline added by Encode.
+	data := buf.Bytes()[:buf.Len()-1]
```

Comment:
It's probably fine to leave the newline

Reply:
You are right. I initially added the newline for consistency with `json.Marshal`. However, I've tested it without the newline, and it functions properly. I've removed it in my new commit.